spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,50 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+      table: DeltaTableV2,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String],
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTable = withActiveSession(sparkSession) {
+    val (targetRelation, targetPath) = if (new Path(target).isAbsolute()) {
+      (UnresolvedRelation(TableIdentifier(target, Some("delta"))), Some(target))
+    } else {
+      val tableIdentifier = sparkSession
+        .sessionState
+        .sqlParser
+        .parseTableIdentifier(target)
+      (UnresolvedRelation(tableIdentifier), None)
+    }
+    val source = DataSourceV2Relation.create(table, None, table.getTableIdentifierIfExists.map(
+      id => Identifier.of(id.database.toArray, id.table)))
+
+    val maybeTimeTravel = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
+      TimeTravel(
+        source,
+        timestampAsOf.map(Literal(_)),
+        versionAsOf,
+        None
+      )
+    } else {
+      source
+    }
+
+    val qe = sparkSession.sessionState.executePlan(CloneTableStatement(maybeTimeTravel,
+      targetRelation, false, replace, true, properties.toMap, targetPath))
+
+    // call `QueryExecution.toRDD` to trigger the execution of commands.
+    SQLExecution.withNewExecutionId(qe, Some("clone delta table"))(qe.toRdd)
+
+    // Return DeltaTable Object.
```

Comment:
```suggestion
```

I don't think we need this comment as the code below is pretty obvious about what is it trying to achieve.

Reply:
Removed, was just copied from table creation